src/main/java/htsjdk/tribble/util/HTTPHelper.java

```diff
@@ -88,30 +88,35 @@ public long getContentLength() throws IOException {
 
     @Override
     public InputStream openInputStream() throws IOException {
-
         HttpURLConnection connection = openConnection();
         return new WrapperInputStream(connection, connection.getInputStream());
     }
 
 
     /**
-     * Open an input stream for the requested byte range.  Its the client's responsibility to close the stream.
+     * Open an InputStream to stream a slice (range) of the resource.  The host server must support
+     * range byte requests and return a 206 response code (partial response).  If it does not an IOException will
+     * be thrown.
+     *
+     * Its the client's responsibility to close the stream.
      *
      * @param start start of range in bytes
      * @param end   end of range ni bytes
      * @return
      * @throws IOException
-     *
-     * @deprecated  since 12/10/14  Will be removed in a future release, as is somewhat fragile
-     * and not used.
      */
     @Override
-    @Deprecated
     public InputStream openInputStreamForRange(long start, long end) throws IOException {
 
         HttpURLConnection connection = openConnection();
         String byteRange = "bytes=" + start + "-" + end;
         connection.setRequestProperty("Range", byteRange);
+
+        if (connection.getResponseCode() != 206) {
```

Comment:
Good check to add if this is important.  I definitely did not know this detail.

Reply:
If servers don't support range requests they are free according to the spec to return the entire file with response code 200.  This is bad when its a 300 gb BAM